kube/src/lib.rs

```diff
@@ -400,4 +420,84 @@ mod test {
         }
         Ok(())
     }
+
+    #[tokio::test]
+    #[ignore] // needs cluster (will create await a pod)
+    #[cfg(all(feature = "runtime"))]
+    async fn pod_can_await_conditions() -> Result<(), Box<dyn std::error::Error>> {
+        use crate::{
+            api::{DeleteParams, PostParams},
+            runtime::wait::{await_condition, conditions, delete::delete_and_finalize, Condition},
+            Api, Client,
+        };
+        use k8s_openapi::api::core::v1::Pod;
+        use std::time::Duration;
+        use tokio::time::timeout;
+
+        let client = Client::try_default().await?;
+        let pods: Api<Pod> = Api::default_namespaced(client);
+
+        // create busybox pod that's alive for at most 30s
+        let p: Pod = serde_json::from_value(serde_json::json!({
+            "apiVersion": "v1",
+            "kind": "Pod",
+            "metadata": { "name": "busybox-kube4" },
+            "spec": {
+                "containers": [{
+                  "name": "busybox",
+                  "image": "busybox:1.34.1",
+                  "command": ["sh", "-c", "sleep 30"],
+                }],
+            }
+        }))?;
+
+        let pp = PostParams::default();
+        match pods.create(&pp, &p).await {
+            Ok(o) => assert_eq!(p.name(), o.name()),
+            Err(crate::Error::Api(ae)) => assert_eq!(ae.code, 409), // if we failed to clean-up
+            Err(e) => return Err(e.into()),                         // any other case if a failure
+        }
+
+        // Watch it phase for a few seconds
+        let is_running = await_condition(pods.clone(), "busybox-kube4", conditions::is_pod_running());
+        let _ = timeout(Duration::from_secs(15), is_running).await?;
+
+
+        // Verify we can get it
+        let pod = pods.get("busybox-kube4").await?;
+        assert_eq!(pod.spec.as_ref().unwrap().containers[0].name, "busybox");
+
+        // Wait for a more complicated condition: ContainersReady AND Initialized
+        // TODO: remove these once we can write these functions generically
+        fn is_containers_ready() -> impl Condition<Pod> {
```

Comment:
```suggestion
        fn are_containers_ready() -> impl Condition<Pod> {
```

Very tiny nit..

Reply:
prefer sticking to `is_` prefix for scanning / conventions, but can make it gramatically correct with `is_each_container_ready`